spec/pg_easy_replicate/orchestrate_spec.rb

```diff
@@ -51,6 +51,20 @@
 
       expect(pg_publications(connection_url: connection_url)).to eq([])
     end
+
+    it "raises an error when user does not have sufficient privileges" do
+      allow(PG::Connection).to receive(:new).and_return(double('PG::Connection', exec: true))
```

Comment:
So, instead of mocking, i was thinking we would use the database helpers to ensure the right permissions on the user and then replicate it in specs, otherwise these situations are only as good as our mocks. 

The rspec suite sort of runs as an integration spec today, which isn't great but provides good coverage.

Reply:
Alright, let me take another look at that.

Reply:
Ok, have updated those test to make better use of the database helpers setup.